index.js

```diff
@@ -102,6 +109,19 @@ const SBP_BASE_SELECTORS = {
   'sbp/filters/selector/add': function (selector: string, filter: TypeFilter) {
     if (!selectorFilters[selector]) selectorFilters[selector] = []
     selectorFilters[selector].push(filter)
+  },
+  'sbp/domains/lock': function (domainNameOrNames: string | string[] | void) {
+    // If no argument was given then locks every known domain.
+    if (domainNameOrNames === undefined) {
+      for (const domain of Object.values(domains)) {
+        domain.locked = true
+      }
+    } else for (const name of typeof domainNameOrNames === 'string' ? [domainNameOrNames] : domainNameOrNames) {
+      if (!domains[name]) {
+        throw new Error(`SBP: unknown or invalid domain name: ${name}`)
```

Comment:
Should probably simplify this error message to `SBP: cannot lock non-existent domain: ${name}`